core/domain/classifier_services.py

```diff
@@ -224,3 +226,117 @@ def delete_classifier(classifier_id):
     classifier_data_model = classifier_models.ClassifierDataModel.get(
         classifier_id)
     classifier_data_model.delete()
+
+
+def get_classifier_training_job_from_model(classifier_training_job_model):
+    """Gets a classifier training job domain object from a classifier
+       training job model.
+
+    Args:
+        classifier_training_job_model: Classifier training job instance in
+            datastore.
+
+    Returns:
+        classifier_training_job: Domain object for the classifier training job.
+    """
+    return classifier_domain.ClassifierTrainingJob(
+        classifier_training_job_model.id,
+        classifier_training_job_model.algorithm_id,
+        classifier_training_job_model.exp_id,
+        classifier_training_job_model.exp_version_when_created,
+        classifier_training_job_model.state_name,
+        classifier_training_job_model.status,
+        classifier_training_job_model.training_data)
+
+def get_classifier_training_job_by_id(job_id):
+    """Gets a classifier training job from a job_id.
+
+    Args:
+        job_id: str. ID of the classifier training job.
+
+    Returns:
+        classifier_training_job: Domain object for the classifier training job.
+
+    Raises:
+        Exception: Entity for class ClassifierTrainingJobModel with id not
+            found.
+    """
+    classifier_training_job_model = (
+        classifier_models.ClassifierTrainingJobModel.get(job_id))
+    classifier_training_job = get_classifier_training_job_from_model(
+        classifier_training_job_model)
+    return classifier_training_job
+
+
+def _create_classifier_training_job(classifier_training_job):
+    """Creates classifier training job model in the datastore given a
+       classifier training job domain object.
+
+    Args:
+        classifier_training_job: Domain object for the classifier training job.
+
+    Returns:
+        job_id: str. ID of the classifier training job.
+    """
+    job_id = classifier_models.ClassifierTrainingJobModel.create(
+        classifier_training_job.algorithm_id, classifier_training_job.exp_id,
+        classifier_training_job.exp_version_when_created,
```

Comment:
@seanlip @anmolshkl . From what I understand, the exp_version_when_Created refers to the version of the exploration for which the classifier was trained (ClassifierTrainingJob instance was created). When the version of an exploration changes, it triggers update_exploration() which in turn triggers the function that checks for re-training.

So, if the re-training turns out false, the exp_version_when_created remains the same while exp_version is incremented. This is fine because we'd like to access the version of the exploration for which the classifier was created and not the latest version (this version might have a changed exploration, but not significant enough for a re-train)/

If the re-training function holds True, we create another job instance. The exp_version_when_created is now updated to the latest version of the exploration. 

So, basically, the exp_verison_when_created is required so that we can access the version of the exploration for which the job was created. I think this is what we need and we shouldn't be updating the version along with the exploration. So, it seems pretty good to me.